app/assets/javascripts/discourse/tests/unit/controllers/admin-user-badges-test.js

```diff
@@ -58,4 +61,59 @@ module("Unit | Controller | admin-user-badges", function (hooks) {
     );
     assert.deepEqual(badgeNames, sortedNames, "sorts badges by name");
   });
+
+  test("performGrantBadge", async function (assert) {
+    const GrantBadgeStub = sinon.stub(UserBadge, "grant");
+    const controller = this.owner.lookup("controller:admin-user-badges");
+    const badgeToGrant = Badge.create({
+      id: 3,
+      name: "A Badge",
+      enabled: true,
+      manually_grantable: true,
+    });
+    const otherBadge = Badge.create({
```

Comment:
Why do you need these badges? Wouldn't just the `badgeToGrant` be enough to test that we're properly passing the `badgeReason`?